### PR TITLE
Don't fail Hibernate Validator when no RESTEasy Reactive request is in progress

### PR DESCRIPTION
**Pull request by geoand**
Relates to: #19834

### CODE REVIEW

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/injection/ContextProducers.java [hunk 1]
```diff
@@ -133,6 +133,10 @@ SecurityContext securityContext() {
     }
 
     private ResteasyReactiveRequestContext getContext() {
-        return CurrentRequestManager.get();
+        ResteasyReactiveRequestContext context = CurrentRequestManager.get();
+        if (context == null) {
+            throw new IllegalStateException("No RESTEasy Reactive request in flight");
```
Comment (gastaldi) [inline independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/injection/ContextProducers.java:138]:
```suggestion
            throw new IllegalStateException("No RESTEasy Reactive request in-flight");
```
See https://cayceberryman.com/hyphens-and-dashes-hyphens/

or maybe more directly: 

```suggestion
            throw new IllegalStateException("No RESTEasy Reactive request in progress");
```

File: extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/ResteasyReactiveContextLocaleResolver.java [hunk 2]
```diff
@@ -18,6 +18,11 @@ public ResteasyReactiveContextLocaleResolver(HttpHeaders headers) {
 
     @Override
     protected HttpHeaders getHeaders() {
-        return headers;
+        try {
+            headers.getLength(); // this forces the creation of the actual object which will fail if there is no request in flight
+            return headers;
+        } catch (IllegalStateException e) {
+            return null;
```
Comment (gsmet) [inline extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/ResteasyReactiveContextLocaleResolver.java:25]: It's not possible to return an empty header object? I'm not sure I understand how the case of not having a request can happen and when the user will be confronted to it?

Comment (gsmet) [reply extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/ResteasyReactiveContextLocaleResolver.java:25]: What I understand from the OP's case is that `ResteasyReactiveRequestContext.getHttpHeaders()` fails somehow. I'm wondering if we should return empty headers there and thus cover this case entirely.

Note that I have absolutely no knowledge of the RR codebase so I'm just thinking out loud, given the pattern above looks hackyish and fixing the issue at the source might be a more general solution.

**Review by gsmet — COMMENTED** [review]
Added some questions.

File: integration-tests/pom.xml [hunk 3]
```diff
@@ -123,6 +123,7 @@
                 <module>class-transformer</module>
                 <module>shared-library</module>
                 <module>hibernate-validator</module>
+                <module>hibernate-validator-resteasy-reactive</module>
```
Comment (gastaldi) [inline integration-tests/pom.xml:126]: Should this new module be listed in https://github.com/quarkusio/quarkus/blob/main/.github/native-tests.json?

Comment (gsmet) [reply integration-tests/pom.xml:126]: Maybe let's not add minutes to the native build if we don't think native will introduce issues.

File: extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/ResteasyReactiveContextLocaleResolver.java (hunk 2 above)
Comment (geoand) [reply extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/ResteasyReactiveContextLocaleResolver.java:25]: `ResteasyReactiveRequestContext.getHttpHeaders()` would indeed make sense to return an empty headers in this case. However it does not make sense for othes in `org.jboss.resteasy.reactive.server.injection.ContextProducers`, so we need a consistent approach

File: integration-tests/pom.xml (hunk 3 above)
Comment (geoand) [reply integration-tests/pom.xml:126]: That was my though exactly for not adding it

**Review by gsmet — APPROVED** [review]
I'm not sure about the pattern as that's typically the type of pattern that could bite us later but I'll trust you on that :).

**Comment by geoand** [comment]
> I'm not sure about the pattern as that's typically the type of pattern that could bite us later but I'll trust you on that :).

You know by know that I'll be the first one to say you were right if the future proves you correct :)

**Pull request merged**